test/TestERC20Firefly.sol

```diff
@@ -0,0 +1,33 @@
+pragma solidity ^0.5.16;
+
+import "./firefly/FireflyCheck.sol";
+import "../contracts/ERC20.sol";
+
+contract TestERC20Symbolic is FireflyCheck {
+    function _testTransfer() public {
+        address owner = this.firefly_genAddress();
+        uint256 initSupply = this.firefly_genUint256();
+
+        ERC20 token = new ERC20("gold", "GLD", owner, initSupply);
+
+        address from  = address(this);
+        address to    = this.firefly_genAddress();
+        uint256 value = this.firefly_genUint256();
+
+        require(from != to);
+
+        uint256 balFrom = token.balanceOf(from);
+        uint256 balTo   = token.balanceOf(to);
+        require(value <= balFrom);
+
+        require(token.transfer(to, value));
+
+        uint256 balFromAfter = token.balanceOf(from);
+        uint256 balToAfter   = token.balanceOf(to);
+
+        this.firefly_assert(balFromAfter <= balFrom);
+        this.firefly_assert(balToAfter >= balTo);
+
+        this.firefly_assert(balFrom - balFromAfter == balToAfter - balTo);
```

Comment:
Let's also just use `assert` here, instead of `this.firefly_assert`. We can intercept `INVALID` opcode just as well as a call to `firefly_assert`, so no need for another function, and it's closer to what devs are used to.